src/leveled_runner.erl

```diff
@@ -301,13 +275,7 @@ tictactree(SnapFun, {Tag, Bucket, Query}, JournalCheck, TreeSize, Filter) ->
 
             % Close down snapshot when complete so as not to hold removed
             % files open
-            ok = leveled_penciller:pcl_close(LedgerSnap),
-            case JournalCheck of
```

Comment:
Now you build the afterfunction as part of the SnapFun.
But it is hard to follow from the code whether indeed you do this JournalCheck or in line 608 DefferedFetch check.
The code gets clearer, but difficult to validate that it has the same behaviour.
In leveled_bookie the return_snap_fun seems to always close both files, whereas code here suggests that that need not always be the case.

Reply:
The tests are comprehensive on this I think, so I'm happy with the risks of the change.  

The only reason for having a JournalSnapshot was to do the JournalCheck.  Previously this was a bit specific ... as a developer I know that for this runner the JournalSnapshot will only present when JournalCheck is true - so test for JournalCheck.  Now with AfterFun/SnapFun it is cleaner that only the Bookie, not the Runner, needs knowledge of whether a snapshot was taken (and therefore if it should be closed).